### ISSUE TITLE
`Optim.jl` Causes a Strange Type Instability in `norm`

### ISSUE THREAD

**Issue by cortner** [JuliaOpt/Optim.jl#283]
``` julia
import Optim
function test_norm(R::Vector{Float64})
#         r = Base.LinAlg.vecnorm2(R)
        r = norm(R)
    return r
end
R1 = rand(3);
@code_warntype test_norm(R1)
```

Output is 

```
Variables:
  #self#::#test_norm
  R::Array{Float64,1}
  r::Any

Body:
  begin
      r::Any = $(Expr(:invoke, LambdaInfo for vecnorm(::Array{Float64,1}, ::Int64), :(Base.LinAlg.vecnorm), :(R), 2)) # line 4:
      return r::Any
  end::Any
```

Comments:
- With `vecnorm2` the type of `r` is inferred correctly. 
- If I don't `import Optim` first, then the type is also inferred correctly.
- This happens on `v0.5-rc4`, but not on `v0.4.6`
- It happens on `Optim` `master` as well as the current release.

**Comment by pkofod** [comment]
I'm thinking https://github.com/JuliaLang/julia/issues/18465 https://github.com/JuliaLang/julia/issues/18449 perhaps? can you try to remove the precompilation line and try again?

**Comment by cortner** [comment]
yes, it is fixed in the nightly.

**Comment by pkofod** [comment]
Well... yeah... Good to know it is fixed at least :)

**Comment by cortner** [comment]
Before closing here, can we wait until it is clear that 0.5 will backport the fix?

**Comment by pkofod** [comment]
Yes, let's keep it open.

**Comment by pkofod** [comment]
Well, there doesn't seem to be any 0.5.x before 0.6.0. It's fixed on v0.6 so I guess I might as well just close this, as v0.6 should be "just around the corner".
